### PR TITLE
Fix heartbeat type and only accept > 0 value

### PR DESCRIPTION
**Pull request by snrism**
This PR addresses the following open issue https://github.com/docker/swarm/issues/558

The changes involve:
+ Updating hearbeat type to be uint64
+ Check if heartbeat value is > 0

### CODE REVIEW

File: discovery/consul/consul.go [hunk 1]
```diff
@@ -40,7 +41,11 @@ func (s *ConsulDiscoveryService) Initialize(uris string, heartbeat int) error {
 		return err
 	}
 	s.client = client
-	s.heartbeat = time.Duration(heartbeat) * time.Second
+	if heartbeat < 1 {
```
Comment (vieux) [inline discovery/consul/consul.go]: the check is already done in `manage.go` and `join.go` I don't think it's necessary to do it here. (it's a lot of duplicated code in every discovery service)

**Comment by vieux** [comment]
@snrism please run `golint -min_confidence=0.9 ./...` and fix the issues, thanks

File: discovery/consul/consul.go (hunk 1 above)
Comment (snrism) [inline discovery/consul/consul.go]: Hmm, sure..will remove that in the discovery service and update it

**Comment by snrism** [comment]
@vieux I removed the check in discovery backends and removed unwanted code. Let me know if this looks OK. 

Thanks,
Sriram

**Comment by snrism** [comment]
@vieux: I combined the commits into a single commit and removed all unwanted code. I did "golint -min_confidence=0.9 ./...", but looks like the CI is failing again. any thoughts?

**Comment by vieux** [comment]
LGTM

**Comment by vieux** [comment]
ping @abronan

**Comment by abronan** [comment]
LGTM

**Pull request merged**